src/cluster.c

```diff
@@ -302,8 +330,8 @@ int clusterLoadConfig(char *filename) {
             }
             n->slaveof = master;
             clusterNodeAddSlave(master,n);
-        }
-
+        } 
+        
```

Comment:
Shouldn't we do some validation that our master has the same shard ID as us here? Which one do we override if they disagree. Based on my previous understanding, we want nodes to be connected to shards and not to specific masters anymore.

Reply:
Agreed. I think I will bail to `fmterr` if the two don't agree on the shard id.